graylog2-web-interface/src/views/components/Query.test.jsx

```diff
@@ -124,14 +121,74 @@ describe('Query', () => {
              queryId="someQueryId"
              showMessages
              allFields={Immutable.List()}
-             fields={Immutable.List()}>
-        Sidebar Content
-      </Query>
+             fields={Immutable.List()} />
     ));
     const widgetGrid = wrapper.find(WidgetGrid);
     expect(widgetGrid).toHaveLength(1);
     expect(widgetGrid).toHaveProp('errors', { widget1: [error1], widget2: [error2] });
     expect(widgetGrid).toHaveProp('data', { widget1: [], widget2: [] });
     expect(widgetGrid).toHaveProp('widgets', { widget1, widget2 });
   });
+
+  it('renders dashboard widget creation explanation on the dashboard page, if no widget is defined', () => {
+    const results = {
+      errors: [],
+      searchTypes: {},
+    };
+    const wrapper = mount((
+      <ViewTypeContext.Provider value={View.Type.Dashboard}>
+        <Query results={results}
+               widgetMapping={widgetMapping}
+               widgets={Immutable.Map()}
+               onToggleMessages={() => {}}
+               queryId="someQueryId"
+               showMessages
+               allFields={Immutable.List()}
+               fields={Immutable.List()} />
+      </ViewTypeContext.Provider>
+    ));
+
+    expect(wrapper.contains('This dashboard has no widgets yet')).toEqual(true);
+  });
+
+
+  it('renders search widget creation explanation on the search page, if no widget is defined', () => {
+    const results = {
+      errors: [],
+      searchTypes: {},
+    };
+    const wrapper = mount((
+      <ViewTypeContext.Provider value={View.Type.Search}>
+        <Query results={results}
+               widgetMapping={widgetMapping}
+               widgets={Immutable.Map()}
+               onToggleMessages={() => {}}
+               queryId="someQueryId"
+               showMessages
+               allFields={Immutable.List()}
+               fields={Immutable.List()} />
+      </ViewTypeContext.Provider>
+    ));
+
+    expect(wrapper.contains('There are no widgets defined to visualize the search result')).toEqual(true);
+  });
+
+  it('renders no widget creation explanation, if there are some widgets defined', () => {
+    const results = {
+      errors: [],
+      searchTypes: {},
+    };
+    const wrapper = shallow((
+      <Query results={results}
+             widgetMapping={widgetMapping}
+             widgets={widgets}
+             onToggleMessages={() => {}}
+             queryId="someQueryId"
+             showMessages
+             allFields={Immutable.List()}
+             fields={Immutable.List()} />
+    ));
+
+    expect(wrapper.contains('You can create a new widget by selecting a widget type')).toEqual(false);
+  });
```

Comment:
Little more context: I used the `mount` function for the first two tests, because a shallow would not render the message, when using the context provider.
I am still using the `shallow` function for the last test, because it is not necessary to use the context provider there.

Reply:
✅